accelerator/core/core.c

```diff
@@ -239,7 +239,10 @@ status_t ta_find_transactions_obj_by_tag(const iota_client_service_t* const serv
     goto done;
   }
 
-  hash243_queue_copy(&obj_req->hashes, txn_res->hashes, hash243_queue_count(txn_res->hashes));
+  int txn_res_len = hash243_queue_count(txn_res->hashes);
+  for (int i = 0; i < FIND_TXN_LIMITATION && i < txn_res_len; i++) {
+    hash243_queue_push(&obj_req->hashes, hash243_queue_at(txn_res->hashes, i));
```

Comment:
could use `hash243_queue_pop` instead, that way we do not need to iterate through queue with `hash243_queue_at` in each loop.